Bisect: 889db62 __GLIBC__

## Maintainer Summary

2026-04-29 19:55:52 UTC

## Bisect
Nov 3, 2023
889db62 __GLIBC__

## Bad Breadcrumb

```
iccDumpProfile.cpp:92:7: warning: '__GLIBC__' is not defined, evaluates to 0 [-Wundef]
   92 | #elif __GLIBC__
      |       ^
```

--- a/Tools/CmdLine/IccDumpProfile/iccDumpProfile.cpp
+++ b/Tools/CmdLine/IccDumpProfile/iccDumpProfile.cpp
@@ -522,8 +522,11 @@ int main(int argc, char* argv[])
         // Number of actual padding bytes between this tag and closest neighbour (or EOF)
         // Should be 0-3 if compliant. Negative number if tags overlap!
         // Guard against unsigned integer overflow in subtraction
-        if (i->TagInfo.offset + (icUInt64Number)i->TagInfo.size > (icUInt64Number)closest)
-            pad = -((int)((icUInt64Number)i->TagInfo.offset + i->TagInfo.size - closest));
+        icUInt64Number tagEnd = (icUInt64Number)i->TagInfo.offset + i->TagInfo.size;
+        if (tagEnd > (icUInt64Number)closest) {
+            icUInt64Number overlap = tagEnd - (icUInt64Number)closest;
+            pad = (overlap > (icUInt64Number)INT_MAX) ? INT_MIN : -(int)overlap;
+        }
         else
             pad = closest - i->TagInfo.offset - i->TagInfo.size;
 

